### DISCUSSION TITLE
Getting error on op-geth and op-node when setting up OP mainnet archive node

### DISCUSSION THREAD

**Discussion by danlix2000** [ethereum-optimism/developers#198] [Node Operators]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I am setting up OP mainnet archive node.  used below flags for op-geth and op-node

`ExecStart=/home/ubuntu/op-geth/build/bin/geth --datadir=/data/datadir --http --http.port=8545 --http.addr=0.0.0.0 --authrpc.addr=localhost --authrpc.jwtsecret=./jwt.txt --verbosity=3 --rollup.sequencerhttp=https://mainnet-sequencer.optimism.io/ --nodiscover --syncmode=full --maxpeers=0  --op-network=op-mainnet --rollup.historicalrpc=http://10.199.11.3:8546 --gcmode archive `

op-node

`ExecStart=/home/ubuntu/optimism/op-node/bin/op-node         --l1=https://ethereum.xxxx.xyz/ethmainnetnvl?apikey=xxxxxx         --l1.rpckind=basic         --l2=ws://localhost:8551         --l2.jwt-secret=./jwt.txt         --network=op-mainnet         --rpc.addr=0.0.0.0         --rpc.port=8547`

### Protocol Description

op-geth version =  v1.101305.0
op-node version = op-node/v1.4.2

### Node Logs

**Op-geth error**

`geth[802120]: Fatal: Failed to register the Ethereum service: database contains incompatible genesis (have d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3, new 7ca38a1916c42007829c55e69d3e9a73265554b586a499015373241b8a3fa48b)`

**op-node error**

` op-node[801494]: t=2024-01-17T16:51:03+0000 lvl=info msg="Rollup Config" l2_chain_id=10 l2_network=OP-Mainnet l1_chain_id=1 l1_network=mainnet l2_start_time=1,686,068,903 l2_block_hash=0xdbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3 l2_block_number=105,235,063 l1_block_hash=0x438335a20d98863a4c0c97999eb2481921ccd28553eac6f913af7c12aec04108 l1_block_number=17,422,590 regolith_time="@ genesis" canyon_time="@ 1704992401 ~ Thu Jan 11 17:00:01 UTC 2024" delta_time="(not configured)" eclipse_time="(not configured)" fjord_time="(not configured)" interop_time="(not configured)"
 op-node[801494]: t=2024-01-17T16:51:03+0000 lvl=info msg="Initializing rollup node" version=v0.0.0-f2229b8b-1703194093
 op-node[801494]: t=2024-01-17T16:51:04+0000 lvl=eror msg="Error initializing the rollup node" err="failed to init L2: failed to get L2 genesis blockhash: failed to determine L2BlockRef of height 105235063, could not get payload: not found"
 op-node[801494]: t=2024-01-17T16:51:04+0000 lvl=crit msg="Application failed"                 message="failed to setup: unable to create the rollup node: failed to init L2: failed to get L2 genesis blockhash: failed to determine L2BlockRef of height 105235063, could not get payload: not found"`

### Additional Information

used latest archive data - https://datadirs.optimism.io/mainnet-2024-01-03.tar.zst

**Comment by imtipi** [comment]
according to https://docs.optimism.io/builders/node-operators/network-upgrades/canyon
On op-node set --rollup.load-protocol-versions to true.
have you set this?

**Reply by danlix2000** [reply]
Now I added that flag to op-node
Now previous error is gone

```
`t=2024-01-18T03:21:03+0000 lvl=info msg="Rollup Config" l2_chain_id=10 l2_network=OP-Mainnet l1_chain_id=1 l1_network=mainnet l2_start_time=1,686,068,903 l2_block_hash=0xdbf6a80fef073de06add9b0d14026d6e5a86c85f6d102c36d3d8e9cf89c2afd3 l2_block_number=105,235,063 l1_block_hash=0x438335a20d98863a4c0c97999eb2481921ccd28553eac6f913af7c12aec04108 l1_block_number=17,422,590 regolith_time="@ genesis" canyon_time="@ 1704992401 ~ Thu Jan 11 17:00:01 UTC 2024" delta_time="(not configured)" eclipse_time="(not configured)" fjord_time="(not configured)" interop_time="(not configured)"
 t=2024-01-18T03:21:03+0000 lvl=info msg="Initializing rollup node" version=v0.0.0-f2229b8b-1703194093
 t=2024-01-18T03:21:03+0000 lvl=warn msg="failed to dial address, but may connect later" addr=ws://localhost:8551
t=2024-01-18T03:21:05+0000 lvl=warn msg="failed to dial address, but may connect later" addr=ws://localhost:8551`
```
So need to fix op-geth error 
`Fatal: Failed to register the Ethereum service: database contains incompatible genesis (have d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3, new 7ca38a1916c42007829c55e69d3e9a73265554b586a499015373241b8a3fa48b)`

**Reply by imtipi** [reply]
i think you have to sync again,since you already have diverging state
